File: 01-_start-manual/elixir_year/lib/elixir_year_web/controllers/year_controller.ex
defmodule ElixirYearWeb.YearController do
  use ElixirYearWeb, :controller

  def get_year do
    years = [2015, 2016, 2017, 2018, 2019, 2020]
    rnd = Enum.random(1..250)
    task = Task.async(fn -> Process.sleep(rnd) end)
    Task.await(task)
    # get a random element from the list of years
    year = Enum.random(years)
  end

  def index(conn, _params) do
    year = get_year()
    render(conn, "index.html", year: year)
  end

end

File: 00-start/java-year/src/main/java/io/honeycomb/examples/javaotlp/YearController.java
package io.honeycomb.examples.javaotlp;

import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.RequestMapping;

import java.util.HashMap;
import java.util.Map;
import java.util.Random;

@RestController
public class YearController {

	private static final int[] YEARS = new int[]{2015, 2016, 2017, 2018, 2019, 2020};
	private static final Random generator = new Random();

	@RequestMapping(value="/", produces="text/html")
	public String index() {
		return "service: <a href='/year'>/year</a>";
	}

	@RequestMapping("/year")
	public Map<String, Object> year() {
		try {
			Thread.sleep(generator.nextInt(250));
		} catch (InterruptedException e) {
			e.printStackTrace();
		}
		Map<String, Object> response = new HashMap<>();
		response.put("language", "Java");
		response.put("year", getYear());

		return response;
	}

	public int getYear() {
		int rnd = generator.nextInt(YEARS.length);
		try {
			Thread.sleep(generator.nextInt(250));
		} catch (InterruptedException e) {
			e.printStackTrace();
		}		
		return YEARS[rnd];
	}
}
